LeaderboardBackend/Controllers/AccountController.cs

```
@@ -18,71 +18,57 @@ public class AccountController(IUserService userService) : ApiController
-        - **Username**:
-          - **UsernameFormat**: Invalid username format
-        - **Password**:
-          - **PasswordFormat**: Invalid password format
-        - **Email**:
-          - **EmailValidator**: Invalid email format
+        A `User` with the specified username already exists. The validation
+        error code `UsernameTaken` will be returned.
         """,
         typeof(ValidationProblemDetails)
     )]
     public async Task<ActionResult<UserViewModel>> Register(
-
         [FromBody, SwaggerRequestBody(
             "The `RegisterRequest` instance from which to register the `User`.",
             Required = true
-        )] RegisterRequest request
-    ,
+        )] RegisterRequest request,
         [FromServices] IAccountConfirmationService confirmationService
     )
     {
-        CreateUserResult result = await userService.CreateUser(request);
+        // Check if we already have a user with the request's email first
+        User? possiblyExistingUser = await userService.GetUserByEmail(request.Email);
 
-        if (result.TryPickT0(out User user, out CreateUserConflicts conflicts))
+        if (possiblyExistingUser is not null)
         {
-            CreateConfirmationResult r = await confirmationService.CreateConfirmationAndSendEmail(user);
+            EmailExistingResult r = await confirmationService.EmailExistingUserOfRegistrationAttempt(possiblyExistingUser);
 
             return r.Match<ActionResult>(
-                confirmation => CreatedAtAction(
-                    nameof(UsersController.GetUserById),
-                    "Users",
-                    new { id = user.Id },
-                    UserViewModel.MapFrom(confirmation.User)
-                ),
-                badRole => StatusCode(StatusCodes.Status500InternalServerError),
+                success => Accepted(),
+                badRole => Accepted(),
                 emailFailed => StatusCode(StatusCodes.Status500InternalServerError)
```

We can just ignore this error I think. Maybe log something internally but don't show the client.